### PR TITLE
Added tf.diag frontend + test

### LINKED ISSUE #13868: diag

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #13868

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by mobley-trent** [comment]
Hi @zaeemansari70, I am requesting a code review 🙂.
I've implemented the test but I am getting this error:

```
FAILED ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py::test_tensorflow_diag[cpu-ivy.functional.backends.numpy-False-False] - AttributeError: 'NoneType' object has no attribute 'backend'
FAILED ivy_tests/test_ivy/test_frontends/test_tensorflow/test_linalg.py::test_tensorflow_diag[cpu-ivy.functional.backends.torch-False-False] - AttributeError: 'NoneType' object has no attribute 'backend'
```

Does this mean the backends are not yet implemented ?

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -253,3 +253,8 @@ def einsum(equation, *inputs, **kwargs):
 @to_ivy_arrays_and_back
 def adjoint(matrix, name=None):
     return ivy.adjoint(matrix)
+
+
+@to_ivy_arrays_and_back
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/tensorflow/linalg.py:258]: Hey @mobley-trent 

The function signature you used here is incorrect I guess given the documentation of this function it indicates that it should be defined as 
```python
def diag(
    diagonal,
    name='diag',
    k=0,
    num_rows=-1,
    num_cols=-1,
    padding_value=0,
    align='RIGHT_LEFT'
):
    pass
```
You can read more here https://www.tensorflow.org/api_docs/python/tf/linalg/diag

**Comment by mobley-trent** [comment]
Hey @hirwa-nshuti I implemented the function with the appropriate parameters but I am getting the same error that I got beforehand

**Comment by AnnaTz** [comment]
Hi @mobley-trent, I tried running your code locally and I don't get the `AttributeError: 'NoneType' object has no attribute 'backend'` that you mentioned. Are you getting the same when you try running other tests as well? If so, there is something wrong with your ide setup. Otherwise, I would suggest syncing with the latest master and trying again then.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @mobley-trent 

Good Job so far,
But there is a mismatch between the shapes of returned array and the ground truth one🙂
Would you look into that?

**Comment by mobley-trent** [comment]
@hirwa-nshuti I actually had a question about the testing. The test case which is failing produces this array:
```
x = array([[0, 0],[0, 0]], dtype=int64)
k = 0
```

But when I test this input on the `diag` function I have implemented and `tf.linalg.diag`, they both produce the same array with shape `(2, 2, 2)` yet the error trace for the test failure indicates that my function returns an array of shape `(2, 3, 3)`. How is this so ? I tested the function in a Jupyter notebook to make sure that both the ground truth function and my implementation produce the same outputs.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @mobley-trent 

Now looks great but the tests for paddle backend are failing, fix this then we can merge this PR.

Great job done😊

**Review by hirwa-nshuti — APPROVED** [review]
Hey @mobley-trent 

Thanks for contributing😊
Going to merge

**Pull request merged**